encord/objects/ontology_labels_impl.py

```diff
@@ -1744,24 +1768,40 @@ def _to_encord_data_unit(self, frame_level_data: FrameLevelImageGroupData) -> Di
             ret["data_link"] = frame_level_data.data_link
 
         ret["data_type"] = frame_level_data.file_type
-
         ret["data_sequence"] = data_sequence
 
-        if self.data_type != DataType.AUDIO:
-            ret["width"] = frame_level_data.width
-            ret["height"] = frame_level_data.height
-
-        else:
+        if self.data_type == DataType.AUDIO:
             ret["audio_codec"] = self._label_row_read_only_data.audio_codec
             ret["audio_sample_rate"] = self._label_row_read_only_data.audio_sample_rate
             ret["audio_bit_depth"] = self._label_row_read_only_data.audio_bit_depth
             ret["audio_num_channels"] = self._label_row_read_only_data.audio_num_channels
+        elif self.data_type == DataType.PLAIN_TEXT or self.data_type == DataType.PDF:
+            pass
+        elif (
+            self.data_type == DataType.IMAGE
+            or self.data_type == DataType.NIFTI
+            or self.data_type == DataType.VIDEO
+            or self.data_type == DataType.IMG_GROUP
+            or self.data_type == DataType.DICOM
+            or self.data_type == DataType.DICOM_STUDY
+        ):
+            ret["width"] = frame_level_data.width
+            ret["height"] = frame_level_data.height
+        elif self.data_type == DataType.MISSING_DATA_TYPE:
+            raise LabelRowError("Label row is missing data type.")
+        else:
+            exhaustive_guard(self.data_type)
 
         ret["labels"] = self._to_encord_labels(frame_level_data)
 
-        if self._label_row_read_only_data.duration is not None:
+        if self._label_row_read_only_data.duration is not None and self.data_type != DataType.PLAIN_TEXT:
```

Comment:
Yea you're right about the original branching condition. Ideally, we only have to check the data_type before adding properties to the `data_unit`, and not have to check whether the value from `_label_row_read_only_data` is `None` or not.

However, we do require this. 
- Because in BE, we do some validation, see `PublicLabelPayload.from_dict`.
- I think we try to use `model_dump(exclude_none=True)` before this validation occurs, see `process_labels_for_post_payload` in `label_helpers.py`, but it doesn't exclude None for nested values in dicts.

In conclusion, sadly, this branching condition is necessary to prevent `duration/fps` from being initialised as a value in this dict, because this dict will be passed to the BE during label updates, and cause it to fail if we get a dict property that we do not expect. I do agree with you that we should find a simpler way to do this. This `to_encord_dict` method is getting pretty unreadable, partly due to my fault as well.

@sergei-encord any thoughts on this?